src/searchengine/nova3/helpers.py

```diff
@@ -89,10 +89,10 @@ def entity2char(m: re.Match[str]) -> str:
     return re.sub(r'&#x(\w+);', lambda x: chr(int(x.group(1), 16)), t)
 
 
-def retrieve_url(url: str, custom_headers: Mapping[str, Any] = {}) -> str:
+def retrieve_url(url: str, custom_headers: Mapping[str, Any] = {}, post_data: Optional[bytes] = None) -> str:
```

Comment:
IMO `post_data` is too specific, other plugins might use it for other purposes.
```suggestion
def retrieve_url(url: str, custom_headers: Mapping[str, Any] = {}, request_data: Optional[bytes] = None) -> str:
```

Reply:
Currently retrieve_url() has no way of specifying an HTTP method. So if (request|post)_data is provided, urllib will perform a POST. Making that variable always and only relevant to POST.

Do you still want to change for request_data?

Reply:
Maybe I should also add `method` (or `request_method`) to retrieve_url?

Reply:
>Do you still want to change for request_data?

Yes.
You regard `post_data` make perfect sense now but what about later when something added other parameters? or extended the usage. That name will not be precise anymore and the worst is you can't change it because it will break someone's script using keyword arguments.

>Maybe I should also add method (or request_method) to retrieve_url?

Does it make sense to provide it to the plugin authors?